lib/Backends/Interpreter/InterpreterNodes.cpp

```diff
@@ -5558,3 +5558,134 @@ void BoundInterpreterFunction::fwdMFCCInst(glow::MFCCInst const *I) {
+  unsigned outputWidth = I->getOutputHeight();
+  unsigned samplingRatio = I->getSamplingRatio();
+  auto boxesH = getTensor(boxes)->getHandle<float>();
+  auto featureMapH = getTensor(featureMap)->getHandle<float>();
+  auto batchIndicesH = getTensor(batchIndices)->getHandle<int64_t>();
+  auto resultH = getTensor(result)->getHandle<float>();
+  getTensor(result)->zero();
+  const unsigned imageHeight = featureMap->dims()[2];
+  const unsigned imageWidth = featureMap->dims()[3];
+  const unsigned numBoxes = result->dims()[0];
+  const unsigned depth = result->dims()[1];
+  for (uint32_t b = 0; b < numBoxes; b++) {
+    const std::vector<float> box = {boxesH.at({b, 0}), boxesH.at({b, 1}),
+                                    boxesH.at({b, 2}), boxesH.at({b, 3})};
+    std::vector<BinGrid> binGrids(samplingRatio * samplingRatio * outputHeight *
+                                  outputWidth);
+    // get the xy coordinates in the resized image(grid)
+    getROIAlignInterpolationCoordinates(imageHeight, imageWidth, outputHeight,
+                                        outputWidth, box, samplingRatio,
+                                        binGrids);
+    const uint32_t batchIndex = batchIndicesH.at({
+        b,
+    });
+    for (uint32_t d = 0; d < depth; ++d) {
+      uint64_t binCount = 0;
+      for (uint32_t H = 0; H < outputHeight; ++H) {
+        for (uint32_t W = 0; W < outputWidth; ++W) {
+          std::vector<float> pixels;
+          for (uint32_t h = 0; h < samplingRatio; ++h) {
+            for (uint32_t w = 0; w < samplingRatio; ++w) {
+              BinGrid bG = binGrids[binCount++];
+              // The four pixels of  the i/p image surrounding the point of
+              // interest (POI) in the resized image
+              const float topLeft =
+                  featureMapH.at({batchIndex, d, bG.Y0, bG.X0});
+              const float topRight =
+                  featureMapH.at({batchIndex, d, bG.Y0, bG.X1});
+              const float bottomLeft =
+                  featureMapH.at({batchIndex, d, bG.Y1, bG.X0});
+              const float bottomRight =
+                  featureMapH.at({batchIndex, d, bG.Y1, bG.X1});
+              // bilinear interpolation = interpolation along {top_horizontal +
+              // bottom_horizontal + vertical} lines or bilinear interpolation =
+              // interpolation along {left_vertical + right_vertical +
+              // horizontal} lines interpolation along top_horizontal line
+              const float top = topLeft + ((topRight - topLeft) * bG.xDiff);
+              // interpolation along bottom_horizontal line
+              const float bottom =
+                  bottomLeft + ((bottomRight - bottomLeft) * bG.xDiff);
+              // interpolation along vertical line
+              pixels.push_back(top + ((bottom - top) * bG.yDiff));
+            } // end of w
+          }   // end of h
+          // {Average or Max} pooling
+          resultH.at({b, d, H, W}) =
```

Comment:
featureMap's and result's layout here is [N,C,H,W] (see question below about layout)

Reply:
I have modified. However, I have kept the loops in the following order **b,H,W,d,h,w** (Previously it was **b, d, H, W, h, w**). By this way I can combine pooling and crop+resize in these loops. Otherwise I have to create one more set of loops for pooling or create two dimensional vector of pixels with second dimension over depth.

Reply:
The ROIAlign node is also modified according to NHWC